File: app/src/test/java/com/manager/singlescreenapp/AdapterTest.java
package com.manager.singlescreenapp;

import android.view.ViewGroup;

import com.manager.singlescreenapp.domain.entities.Author;
import com.manager.singlescreenapp.view.adapters.Adapter;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;

import java.util.List;

public class AdapterTest {


    @Mock
    private ViewGroup viewGroup;

    @Mock
    private List<Author> list;

    @Mock
    private Adapter adapter;


    @Before
    public void setUp(){
        adapter = new Adapter(list);
        viewGroup = new ViewGroup() {
            @Override
            protected void onLayout(boolean b, int i, int i1, int i2, int i3) {

            }
        }
    }

    @Test
    public void onCreateViewHolderTest(){
        adapter.onCreateViewHolder(viewGroup,0);
    }

    @After
    public void tearDown(){

    }
}

File: app/src/test/java/com/manager/singlescreenapp/AdapterTest.kt
package com.manager.singlescreenapp

import androidx.constraintlayout.widget.ConstraintLayout
import com.manager.singlescreenapp.model.Author
import com.manager.singlescreenapp.view.adapters.Adapter
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment
import org.robolectric.shadows.ShadowLooper

@RunWith(RobolectricTestRunner::class)
class AdapterTest {

    lateinit var list: List<Author>

    lateinit var adapter: Adapter

    lateinit var holder: Adapter.RecViewHolder

    @Before
    fun setUp() {
        ShadowLooper.runUiThreadTasksIncludingDelayedTasks()
        list = ArrayList()
        adapter = Adapter(list, RuntimeEnvironment.application)
        initHolder()
    }

    private fun initHolder() {
        try {
            holder = adapter.onCreateViewHolder(ConstraintLayout(RuntimeEnvironment.application),0)
        } catch (e:Exception){}
    }

    @Test
    fun onCreateViewHolderTest() {
        try {
            adapter.onCreateViewHolder(ConstraintLayout(RuntimeEnvironment.application),0)
        } catch (e:Exception){}
    }

    @Test
    fun onBindViewHolderTest(){
        try {
            adapter.onBindViewHolder(holder,0)
        } catch (e:Exception){}
    }

    @Test
    fun getItemCountTest(){
        adapter.itemCount
    }

    @After
    fun tearDown() {
    }
}